### DISCUSSION TITLE
Detect if value is optimistic

### DISCUSSION THREAD

**Discussion by alexnsorensen** [urql-graphql/urql#3662] [Q&A]
I have optimistic updates working great, however, is there a way to determine if a value is optimisitc or not?

My current situation is this:

1. User Adds item to list
2. Item gets added optimistically and appears instantly (Request 1)
3. User decides to delete item
4. Item gets removed optimisitally (Request 2)
5. Request 1 returns and add the item back to the list
6. Request 2 fails as an invalid ID was sent to the server

What I'm after is stopping the user from being able to delete said item until it's back from the server, my only solution is to inject a property into the optimistic value.

**Comment by DeeDeeQQ** [comment]
You also able to write your own exchange to hold Request 2 and wait for the response of Request 1. Place it somewhere between cacheExchange & fetchExchange and just filter requests that has invalid ID. Inside the cache exchange you could get it from `info.optimistic`
